tests/manage/mcg/test_namespace_crd.py

```diff
@@ -358,6 +365,362 @@ def test_distribution_of_objects_in_ns_bucket_crd(
+    def test_list_cached_objects(
+        self, bucket_factory, mcg_obj, cld_mgr, awscli_pod, bucketclass_dict
+    ):
+        """
+        Test list all cached objects in cached namespace bucket.
+        """
+
+        # Create the cached namespace bucket on top of the namespace resource
+        bucket_obj = bucket_factory(
+            amount=1,
+            interface=bucketclass_dict["interface"],
+            bucketclass=bucketclass_dict,
+        )[0]
+        # Upload files to NS bucket
+        self.write_files_to_pod_and_upload(
+            mcg_obj, awscli_pod, bucket_to_write=bucket_obj.name, amount=3
+        )
+        if not check_cached_objects(mcg_obj, bucket_obj.name, 3):
+            raise UnexpectedBehaviour("Object was not cached properly")
+
+    @tier1
+    @pytest.mark.parametrize(
+        argnames=["bucketclass_dict"],
+        argvalues=[
+            pytest.param(
+                {
+                    "interface": "OC",
+                    "namespace_policy_dict": {
+                        "type": "Cache",
+                        "ttl": 100,
+                        "namespacestore_dict": {
+                            "aws": [(1, "eu-central-1")],
+                        },
+                    },
+                    "placement_policy": {
+                        "tiers": [
+                            {"backingStores": [constants.DEFAULT_NOOBAA_BACKINGSTORE]}
+                        ]
+                    },
+                }
+            ),
+        ],
+        ids=[
+            "AWS-OC-Cache",
+        ],
+    )
+    def test_delete_cached_object(
+        self, bucket_factory, mcg_obj, cld_mgr, awscli_pod, bucketclass_dict
+    ):
+        """
+        Test list all cached objects in cached namespace bucket.
+        """
+
+        # Create the cached namespace bucket on top of the namespace resource
+        bucket_obj = bucket_factory(
+            amount=1,
+            interface=bucketclass_dict["interface"],
+            bucketclass=bucketclass_dict,
+        )[0]
+        # Upload files to NS bucket
+        self.write_files_to_pod_and_upload(
+            mcg_obj, awscli_pod, bucket_to_write=bucket_obj.name, amount=1
+        )
+        if not check_cached_objects(mcg_obj, bucket_obj.name, 1):
+            raise UnexpectedBehaviour("Object was not cached properly")
+
+        # Delete the object from mcg interface
+        s3_delete_object(mcg_obj, bucket_obj.name, "testfile0.txt")
+        if not check_cached_objects(mcg_obj, bucket_obj.name, 0):
+            raise UnexpectedBehaviour("Object was not deleted from cache properly")
+
+        # Check deletion in the cloud provider
+        aws_target_bucket = bucket_obj.bucketclass.namespacestores[0].uls_name
+        aws_obj_list = len(
+            list(cld_mgr.aws_client.client.Bucket(aws_target_bucket).objects.all())
+        )
+        if aws_obj_list != 0:
+            raise UnexpectedBehaviour("Object was not deleted from cache properly")
```

Comment:
I believe this logic might be prone to breaking in the future - can you please change the verification to check that the particular object name is not stored, instead of comparing lengths?